tests/test_simulate.py

```diff
@@ -195,36 +194,6 @@ def test_classifiers():
     assert len(list_classifiers()) >= 7
 
 
-def test_partial_simulation(tmpdir):
-
-    # copy state file to tmp dir for changes
-    tmp_h5_state_fp = Path(tmpdir, "tmp_state.h5")
-    copyfile(H5_STATE_FILE, tmp_h5_state_fp)
-
-    check_model(data_fp=DATA_FP_PARTIAL,
-                state_file=tmp_h5_state_fp,
-                n_prior_included=1,
-                n_prior_excluded=1,
-                prior_idx=None,
-                state_checker=check_partial_state)
-
-
-@pytest.mark.xfail(
-    raises=ValueError,
-    reason="prior_idx not available for partly labeled data"
-)
-def test_partial_simulation_2(tmpdir):
```

Comment:
Is there an informative error now if the user uploads a partially labeled dataset? What error is raised?

Reply:
There is indeed a ValueError thrown by `simulate.py`:
https://github.com/asreview/asreview/blob/fee139f2dbc21689488ec59467fbe73523576213/asreview/review/simulate.py#L89

For verification, I ran a test with our representative demo dataset [generic_partial_labels.csv](https://github.com/asreview/asreview/blob/fee139f2dbc21689488ec59467fbe73523576213/tests/demo_data/generic_partial_labels.csv) :
```
% asreview simulate tests/demo_data/generic_partial_labels.csv --state_file myreview.h5 

            _____ _____            _
     /\    / ____|  __ \          (_)
    /  \  | (___ | |__) |_____   ___  _____      __
   / /\ \  \___ \|  _  // _ \ \ / / |/ _ \ \ /\ / /
  / ____ \ ____) | | \ \  __/\ V /| |  __/\ V  V /
 /_/    \_\_____/|_|  \_\___| \_/ |_|\___| \_/\_/

---------------------------------------------------------------------------------
|                                                                                |
|  Welcome to the ASReview Automated Systematic Review software.                 |
|  In this mode the computer will simulate how well the ASReview software        |
|  could have accelerate the systematic review of your dataset.                  |
|  You can sit back and relax while the computer runs this simulation.           |
|                                                                                |
|  GitHub page:        https://github.com/asreview/asreview                      |
|  Questions/remarks:  asreview@uu.nl                                            |
|                                                                                |
---------------------------------------------------------------------------------

Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniforge/base/envs/asreview_m1/bin/asreview", line 33, in <module>
    sys.exit(load_entry_point('asreview', 'console_scripts', 'asreview')())
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/__main__.py", line 80, in main
    entry.load()().execute(sys.argv[2:])
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/entry_points/simulate.py", line 48, in execute
    review_simulate(path, **args_dict)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 314, in review_simulate
    review(dataset, *args, mode='simulate', **kwargs)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 268, in review
    reviewer = get_reviewer(*args, mode=mode, model=model, **kwargs)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 229, in get_reviewer
    reviewer = ReviewSimulate(as_data,
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/simulate.py", line 89, in __init__
    raise ValueError("Expected fully labeled dataset.")
ValueError: Expected fully labeled dataset.
```

Furthermore, I ran a test on a fully labeled benchmark dataset [van_de_Schoot_2017.csv](https://github.com/asreview/systematic-review-datasets/blob/0909240fec2602c3ecef82d131790d390edb33bf/datasets/van_de_Schoot_2017/output/van_de_Schoot_2017.csv):
```
% asreview simulate van_de_Schoot_2017.csv --state_file myreview.h5                    

            _____ _____            _
     /\    / ____|  __ \          (_)
    /  \  | (___ | |__) |_____   ___  _____      __
   / /\ \  \___ \|  _  // _ \ \ / / |/ _ \ \ /\ / /
  / ____ \ ____) | | \ \  __/\ V /| |  __/\ V  V /
 /_/    \_\_____/|_|  \_\___| \_/ |_|\___| \_/\_/

---------------------------------------------------------------------------------
|                                                                                |
|  Welcome to the ASReview Automated Systematic Review software.                 |
|  In this mode the computer will simulate how well the ASReview software        |
|  could have accelerate the systematic review of your dataset.                  |
|  You can sit back and relax while the computer runs this simulation.           |
|                                                                                |
|  GitHub page:        https://github.com/asreview/asreview                      |
|  Questions/remarks:  asreview@uu.nl                                            |
|                                                                                |
---------------------------------------------------------------------------------

The following records are prior knowledge:

2616 - Six years after the wave. Trajectories of posttraumatic stress following a nat..   Johannesson, K. B., Arinell, H., Arnbe..
6075 - Post-traumatic stress disorder and functioning and quality of life outcomes in..   Zatzick, D. F., Weiss, D. S., Marmar, ..
```
And then removed one label manually, which gave the expected result:
```
% asreview simulate van_de_Schoot_2017.csv --state_file myreview.h5

            _____ _____            _
     /\    / ____|  __ \          (_)
    /  \  | (___ | |__) |_____   ___  _____      __
   / /\ \  \___ \|  _  // _ \ \ / / |/ _ \ \ /\ / /
  / ____ \ ____) | | \ \  __/\ V /| |  __/\ V  V /
 /_/    \_\_____/|_|  \_\___| \_/ |_|\___| \_/\_/

---------------------------------------------------------------------------------
|                                                                                |
|  Welcome to the ASReview Automated Systematic Review software.                 |
|  In this mode the computer will simulate how well the ASReview software        |
|  could have accelerate the systematic review of your dataset.                  |
|  You can sit back and relax while the computer runs this simulation.           |
|                                                                                |
|  GitHub page:        https://github.com/asreview/asreview                      |
|  Questions/remarks:  asreview@uu.nl                                            |
|                                                                                |
---------------------------------------------------------------------------------

Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniforge/base/envs/asreview_m1/bin/asreview", line 33, in <module>
    sys.exit(load_entry_point('asreview', 'console_scripts', 'asreview')())
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/__main__.py", line 80, in main
    entry.load()().execute(sys.argv[2:])
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/entry_points/simulate.py", line 48, in execute
    review_simulate(path, **args_dict)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 314, in review_simulate
    review(dataset, *args, mode='simulate', **kwargs)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 268, in review
    reviewer = get_reviewer(*args, mode=mode, model=model, **kwargs)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 229, in get_reviewer
    reviewer = ReviewSimulate(as_data,
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/simulate.py", line 89, in __init__
    raise ValueError("Expected fully labeled dataset.")
ValueError: Expected fully labeled dataset.
```

Reply:
Oke looks good.